uofw/uofw
# Add Chkreg module

## Description
This PR adds the fully reverse-engineered `chkreg` module.

### Highlights
As part of RE'ing this module, two NIDs were cracked:
```
0x7939C851 = sceChkregGetPspModel
0x6894A027 = sceChkregGetPsFlags
```

This leaves only one remaining NID for chkreg to be cracked.

## Motivation and Context


* Shows the structure of the `ConsoleId` 16-byte value to enable debug features in the PSP system ("Debug Settings" in XMB)

## How Has This Been Tested?
The module has been compiled and tested successfully on a PSP-1000. Testing included the following:
* Verify that the PSP still boots and runs correctly (XMB & game)
* Verify that calling chkreg APIs does not crash the system (sceChkregGetPsCode(), sceChkregGetPsFlags(), sceChkregGetPspModel(), _sceChkregCheckRegion())
* Verify that a UMD game disc can be started successfully

## Special Thanks
* @joel16 for his initial work on this module 
* @artart78 for running the module on his PSP to see if it works
* @sajattack for his [nidcracker](https://psp.re/blog/nid-cracking/) program used to crack these NIDs
* @CelesteBlue-dev for the [PS-ConsoleId-wiki](https://github.com/CelesteBlue-dev/PS-ConsoleId-wiki) which provided several data structures and macros used in this RE
* @zecoxao for explaining to me the terms ConsoleId/IPDS and telling me about the structure of IDStorage certificates
